File: python/courtlistener/api/sources.py
"""
Sources API module for CourtListener SDK.
"""

from typing import Dict, Any, Optional, List
from .base import BaseAPI


class SourcesAPI(BaseAPI):
    """API for accessing data sources information."""
    
    def __init__(self, client):
        super().__init__(client)
        self.endpoint = "sources/"
    
    def list(self, **kwargs) -> Dict[str, Any]:
        """
        List data sources.
        
        Args:
            **kwargs: Query parameters
        
        Returns:
            API response with paginated list of sources
        """
        return self.client.get(self.endpoint, params=kwargs)
    
    def get(self, source_id: int) -> Dict[str, Any]:
        """
        Get a specific source by ID.
        
        Args:
            source_id: Source ID
        
        Returns:
            Source data
        """
        return self.client.get(f"{self.endpoint}{source_id}/")
    
    def paginate(self, **kwargs):
        """
        Get paginated sources.
        
        Args:
            **kwargs: Query parameters
        
        Returns:
            PageIterator for sources
        """
        return self.client.paginate(self.endpoint, params=kwargs)

File: php/src/CourtListener/Api/Sources.php
<?php

namespace CourtListener\Api;

use CourtListener\CourtListenerClient;
use CourtListener\Exceptions\CourtListenerException;

/**
 * Sources API endpoint
 */
class Sources extends BaseApi
{
    protected string $endpoint = 'sources/';

    /**
     * Constructor
     *
     * @param CourtListenerClient $client
     */
    public function __construct(CourtListenerClient $client)
    {
        parent::__construct($client);
    }

    /**
     * List sources with advanced filtering
     *
     * @param array $params Query parameters
     * @return array
     * @throws CourtListenerException
     */
    public function listSources(array $params = []) {
        return $this->list($params);
    }

    /**
     * Get a specific sources by ID
     *
     * @param int|string $id Sources ID
     * @param array $params Query parameters
     * @return array
     * @throws CourtListenerException
     */
    public function getSources($id, array $params = []) {
        return $this->get($id, $params);
    }

    /**
     * Search sources
     *
     * @param array $params Search parameters
     * @return array
     * @throws CourtListenerException
     */
    public function searchSources(array $params = []) {
        return $this->search($params);
    }
}
